client/blocks/comment-detail/docs/comment-faker.jsx

```diff
@@ -28,59 +28,107 @@ export const CommentFaker = WrappedCommentList => class extends Component {
 		}
 	}
 
-	deleteCommentPermanently = commentId => this.setState( { comments: omit( this.state.comments, commentId ) } );
+	deleteCommentPermanently = comment => this.doAction( [ comment ], { status: 'delete' } );
 
-	getCommentsFromProps = ( { comments } ) => this.setState( { comments: keyBy( comments, 'ID' ) } );
+	/**
+	 * Sets a status and/or a like value to a list of comments.
+	 *
+	 * `comments` is the array of comments that will receive the action.
+	 * It can contain complete comment objects, but in fact it only needs their (current) `ID`, `i_like`, and `status` properties
+	 * in order to perform combined operations (e.g. disapproving a comment also removes its like).
+	 *
+	 * `action` can contain the new `i_like` and/or `status` value.
+	 *
+	 * @param {Array}  comments
+	 * @param {int}    comments[].ID
+	 * @param {bool}   comments[].i_like
+	 * @param {string} comments[].status
+	 * @param {Object} action
+	 * @param {bool}   action.i_like
+	 * @param {strint} action.status
+	 */
+	doAction = ( comments, { i_like, status } ) => {
+		if ( 'delete' === status ) {
+			return this.setState( {
+				comments: omit( this.state.comments, map( comments, comment => comment.ID ) ),
+			} );
+		}
 
-	setCommentLike = ( commentId, likeValue ) => {
-		const comment = this.state.comments[ commentId ];
-
-		// If like changes to true, also approve the comment
-		this.setState( {
-			comments: {
-				...this.state.comments,
-				[ commentId ]: {
-					...comment,
-					i_like: likeValue,
-					status: likeValue ? 'approved' : comment.status,
-				}
-			},
-		} );
-	}
+		const editedComments = keyBy( map( comments, comment => {
+			// If the comment is not approved anymore, also remove the like, otherwise keep its previous value
+			const newLikeValue = isNil( i_like )
+				? 'approved' === status ? comment.i_like : false
+				: i_like;
```

Comment:
Not a fan as well, but I've kept it (here and for `newStatusValue`) just because it's the faker.

The idea is that we have two properties `foo` and `bar` and we operate like this:
- if `foo` exists → use it
- otherwise
  - if `bar` is some "trigger" value, use `previousFoo`
  - else invert `foo`

I guess I could add some parens to improve readability though.